src/pretix/base/timeframes.py

```diff
@@ -0,0 +1,404 @@
+#
+# This file is part of pretix (Community Edition).
+#
+# Copyright (C) 2014-2020 Raphael Michel and contributors
+# Copyright (C) 2020-2021 rami.io GmbH and contributors
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General
+# Public License as published by the Free Software Foundation in version 3 of the License.
+#
+# ADDITIONAL TERMS APPLY: Pursuant to Section 7 of the GNU Affero General Public License, additional terms are
+# applicable granting you additional permissions and placing additional restrictions on your usage of this software.
+# Please refer to the pretix LICENSE file to obtain the full terms applicable to this work. If you did not receive
+# this file, see <https://pretix.eu/about/en/license>.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+# warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License along with this program.  If not, see
+# <https://www.gnu.org/licenses/>.
+#
+import calendar
+from datetime import date, datetime, time, timedelta
+from itertools import groupby
+from typing import Optional, Tuple
+
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.formats import date_format
+from django.utils.timezone import make_aware, now
+from django.utils.translation import gettext_lazy, pgettext_lazy
+
+from pretix.helpers.daterange import daterange
+
+
+def _quarter_start(ref_d):
+    return ref_d.replace(day=1, month=1 + (ref_d.month - 1) // 3 * 3)
+
+
+def _week_start(ref_d):
+    return ref_d - timedelta(ref_d.weekday())
+
+
+REPORTING_DATE_TIMEFRAMES = (
+    # (identifier, label, start_inclusive, end_inclusive, includes_future, optgroup, describe)
+    (
+        'days_today',
+        pgettext_lazy('reporting_timeframe', 'Today'),
+        lambda ref_d: ref_d,
+        lambda ref_d, start_d: start_d,
+        False,
+        pgettext_lazy('reporting_timeframe', 'by day'),
+        daterange
+    ),
+    (
```

Comment:
What about 'yesterday'? Might be interesting every morning to report what happened the day before …

Reply:
Fair, will add